mdp/nodes/sfa_nodes.py

```diff
@@ -369,6 +371,66 @@ def get_eta_values(self, t=1):
         return self._refcast(t / (2 * numx.pi) * numx.sqrt(self.d))
 
 
+class UnevenlySampledSFANode(SFANode):
+
+    def __init__(self, input_dim=None, output_dim=None, dtype=None,
+                 include_last_sample=True, rank_deficit_method='none'):
+        super(SFANode, self).__init__(input_dim, output_dim, dtype)
+
+        self._include_last_sample = include_last_sample
+
+        # init two covariance matrices
+        # one for the input data
+        self._cov_mtx = UnevenlySampledCovarianceMatrix(dtype)
+        # one for the derivatives
+        self._dcov_mtx = UnevenlySampledCovarianceMatrix(dtype)
+
+        # set routine for eigenproblem
+        self.set_rank_deficit_method(rank_deficit_method)
+        self.rank_threshold = 1e-12
+        self.rank_deficit = 0
+
+        # SFA eigenvalues and eigenvectors, will be set after training
+        self.d = None
+        self.sf = None  # second index for outputs
+        self.avg = None
+        self._bias = None  # avg multiplied with sf
+        self.tlen = None
+
+    def time_derivative(self, x, dt):
+        """
+        Compute the linear approximation of the time derivative
+
+        :param x: The time series data.
+        :type x: numpy.ndarray
+
+        :returns: Piecewise linear approximation of the time derivative.
+        :rtype: numpy.ndarray
+        """
+        # Improvements can be made, by interpolating polynomials
+        return (x[1:, :]-x[:-1, :])/dt[:,None]
+    
+    def _train(self, x, dt, include_last_sample=None):
```

Comment:
As `dt` is not optional, do you want to include an error raise? Otherwise we could set `dt= numx.ones(..)`.

Reply:
I would prefer to have `dt` optional in that sense. If the only way is to defaultt it to `dt= numx.ones(..)` then be it, but wouldn't it be better to allow `dt == None` all the way through, indicating that the normalization `whatever \= dt` can be skipped? This would avoid useless division through one in the fallback case (maybe numpy would optimize it away under the hood but I wouldn't count on it).

Reply:
I think, we then would have to call the covariance methods with `dt=None` and within these we would need a handling of this input which would be non trivial. We could also call the standard covariance methods - which would be the standard SFA, but I believe the most straight forward way would be setting `dt=numx.ones(..)` and inputting that into the covariance function. Especially considering that the unevenly sampled covariance called with `dt=numx.ones(..)` is not exactly the same as the standard covariance.

Reply:
I looked into [`update`](https://github.com/mdp-toolkit/mdp-toolkit/pull/48/files#diff-c45ae376d2565472569ce09e2840b995R169) and think it's not too difficult to make it work with `dt==None`:
At the beginning, set
```
_dt = 1. if dt is None else dt[:, None]
```
then later always use `_dt` in place of `dt` or `dt[:, None]`. This even improves readability as it hides `[:, None]`. Using a plain `1.` should go through with numpy broadcasting and is already cheaper than creating `ones`. I think `update` is the only place that requires further adjustment, or do I overlook something?

Reply:
Thanks for the suggestion, the solutions I had in mind were dirtier. I implemented this and tests run fine (up to the randomized tests  - which not included in this PR -  that sometimes fail due to badly set precision. Next I will adapt the signature of the UnSFA.